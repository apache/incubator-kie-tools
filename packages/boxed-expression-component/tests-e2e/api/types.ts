import {
  DMN15__tConditional,
  DMN15__tContext,
  DMN15__tDecisionTable,
  DMN15__tFilter,
  DMN15__tFor,
  DMN15__tFunctionDefinition,
  DMN15__tInvocation,
  DMN15__tList,
  DMN15__tLiteralExpression,
  DMN15__tQuantified,
  DMN15__tRelation,
} from "@kie-tools/dmn-marshaller/dist/schemas/dmn-1_5/ts-gen/types";

export type BoxedLiteral = { __$$element: "literalExpression" } & DMN15__tLiteralExpression;
export type BoxedRelation = { __$$element: "relation" } & DMN15__tRelation;
export type BoxedContext = { __$$element: "context" } & DMN15__tContext;
export type BoxedDecisionTable = { __$$element: "decisionTable" } & DMN15__tDecisionTable;
export type BoxedList = { __$$element: "list" } & DMN15__tList;
export type BoxedInvocation = { __$$element: "invocation" } & DMN15__tInvocation;
export type BoxedFunction = { __$$element: "functionDefinition" } & DMN15__tFunctionDefinition;
export type BoxedFor = { __$$element: "for" } & DMN15__tFor;
export type BoxedEvery = { __$$element: "every" } & DMN15__tQuantified;
export type BoxedSome = { __$$element: "some" } & DMN15__tQuantified;
export type BoxedConditional = { __$$element: "conditional" } & DMN15__tConditional;
export type BoxedFilter = { __$$element: "filter" } & DMN15__tFilter;

export type Normalized<T> = WithRequiredDeep<T, "@_id">;

type WithRequiredDeep<T, K extends keyof any> = T extends undefined
  ? T
  : T extends Array<infer U>
    ? Array<WithRequiredDeep<U, K>>
    : { [P in keyof T]: WithRequiredDeep<T[P], K> } & (K extends keyof T
        ? { [P in K]-?: NonNullable<WithRequiredDeep<T[P], K>> }
        : T);
