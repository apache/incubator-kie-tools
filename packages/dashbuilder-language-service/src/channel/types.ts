/*
 * Copyright 2022 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { TextDocument } from "vscode-languageserver-textdocument";
import { CompletionItemKind, Range } from "vscode-languageserver-types";

export declare type DashbuilderLsNodeType = "object" | "array" | "property" | "string" | "number" | "boolean" | "null";
export declare type DashbuilderJsonPath = (string | number)[];

/**
 * The AST node used in the LanguageServices
 */
export type DashbuilderLsNode = {
  type: DashbuilderLsNodeType;
  value?: any;
  offset: number;
  length: number;
  colonOffset?: number;
  parent?: DashbuilderLsNode;
  children?: DashbuilderLsNode[];
};

export interface TranslateArgs {
  completion: object | string;
  completionItemKind: CompletionItemKind;
  currentNodeRange?: Range;
  cursorOffset: number;
  document: TextDocument;
  overwriteRange?: Range;
}
