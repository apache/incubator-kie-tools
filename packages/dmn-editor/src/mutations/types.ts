import { NodeType } from "../diagram/connections/graphStructure";
import { NODE_TYPES } from "../diagram/nodes/NodeTypes";

export enum NodeNature {
  DRG_ELEMENT = "DRG_ELEMENT",
  ARTIFACT = "ARTIFACT",
}

export const nodeNatures: Record<NodeType, NodeNature> = {
  [NODE_TYPES.inputData]: NodeNature.DRG_ELEMENT,
  [NODE_TYPES.decision]: NodeNature.DRG_ELEMENT,
  [NODE_TYPES.bkm]: NodeNature.DRG_ELEMENT,
  [NODE_TYPES.knowledgeSource]: NodeNature.DRG_ELEMENT,
  [NODE_TYPES.decisionService]: NodeNature.DRG_ELEMENT,
  [NODE_TYPES.textAnnotation]: NodeNature.ARTIFACT,
  [NODE_TYPES.group]: NodeNature.ARTIFACT,
};
