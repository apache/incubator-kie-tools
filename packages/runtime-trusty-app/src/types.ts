/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
export interface TrustyContextValue {
  config: {
    counterfactualEnabled: boolean;
    explanationEnabled: boolean;
    serverRoot: string;
    basePath: string;
    useHrefLinks: boolean;
  };
}

export type RemoteData<E, D> =
  | { status: RemoteDataStatus.NOT_ASKED }
  | { status: RemoteDataStatus.LOADING }
  | { status: RemoteDataStatus.FAILURE; error: E }
  | { status: RemoteDataStatus.SUCCESS; data: D };

export enum RemoteDataStatus {
  NOT_ASKED,
  LOADING,
  FAILURE,
  SUCCESS,
}

export interface Execution {
  executionId: string;
  executionDate: string;
  executedModelName: string;
  executionType: string;
  executionSucceeded: boolean;
  executorName: string;
}

export interface Executions {
  headers: Execution[];
  total: number;
}

export interface ExecutionRouteParams {
  executionId: string;
  executionType: string;
}

export interface ItemObject {
  name: string;
  value: ItemObjectValue;
}

export type ItemObjectValue = ItemObjectUnit | ItemObjectCollection | ItemObjectStructure;

export interface ItemObjectUnit {
  kind: "UNIT";
  type: string;
  value: string | number | boolean | Array<string | number | boolean> | null;
}

export interface ItemObjectCollection {
  kind: "COLLECTION";
  type: string;
  value: ItemObjectValue[] | null;
}

export type ItemObjectMap = { [key: string]: ItemObjectValue };

export interface ItemObjectStructure {
  kind: "STRUCTURE";
  type: string;
  value: ItemObjectMap;
}

export interface InputRow {
  inputLabel: string;
  inputValue?: ItemObject["value"];
  key: string;
  category: string;
}

export enum evaluationStatus {
  EVALUATING = "Evaluating",
  FAILED = "Failed",
  NOT_EVALUATED = "Not evaluated",
  SKIPPED = "Skipped",
  SUCCEEDED = "Succeeded",
}

export type evaluationStatusStrings = keyof typeof evaluationStatus;

export interface Outcome {
  outcomeId: string;
  outcomeName: string;
  outcomeResult: ItemObjectValue;
  evaluationStatus: evaluationStatusStrings;
  hasErrors: boolean;
  messages: string[];
}

export interface FeatureScores {
  featureName: string;
  featureScore: number;
}

export interface Saliency {
  outcomeId: string;
  featureImportance: FeatureScores[];
}

export enum SaliencyStatus {
  SUCCEEDED = "SUCCEEDED",
  FAILED = "FAILED",
}

export type SaliencyStatusStrings = keyof typeof SaliencyStatus;

export interface Saliencies {
  status: SaliencyStatusStrings;
  statusDetail: string;
  saliencies: Saliency[];
}

export interface ServiceIdentifier {
  groupId?: string;
  artifactId?: string;
  version?: string;
}

export interface ModelData {
  deploymentDate?: string;
  modelId?: string;
  name: string;
  namespace: string;
  modelVersion: string;
  dmnVersion: string;
  serviceIdentifier: ServiceIdentifier;
  model: string;
}

export interface CFSearchInput {
  name: string;
  value: CFSearchInputValue;
}

export type CFSearchInputValue = CFSearchInputUnit | CFSearchInputCollection | CFSearchInputStructure;

export interface CFSearchInputUnit {
  kind: "UNIT";
  type: string;
  fixed?: boolean;
  domain?: CFNumericalDomain | CFCategoricalDomain;
  originalValue: ItemObject["value"];
}

export interface CFNumericalDomain {
  type: "RANGE";
  lowerBound?: number;
  upperBound?: number;
}

export interface CFCategoricalDomain {
  type: "CATEGORICAL";
  categories: string[];
}

export interface CFSearchInputCollection {
  kind: "COLLECTION";
  type: string;
  value: Array<CFSearchInputValue> | null;
}

export type CFSearchInputValueMap = { [key: string]: CFSearchInputValue };

export interface CFSearchInputStructure {
  kind: "STRUCTURE";
  type: string;
  value: CFSearchInputValueMap | null;
}

export enum CFGoalRole {
  UNSUPPORTED,
  ORIGINAL,
  FIXED,
  FLOATING,
}

export interface CFGoal {
  id: string;
  name: string;
  role: CFGoalRole;
  value: ItemObjectValue;
  originalValue: CFGoal["value"];
}

export type CFResult = Array<unknown>;

export interface CFStatus {
  isDisabled: boolean;
  executionStatus: CFExecutionStatus;
  lastExecutionTime: null | string;
}

export enum CFExecutionStatus {
  COMPLETED,
  RUNNING,
  NOT_STARTED,
  FAILED,
  NO_RESULTS,
}

export type CFAnalysisResetType = "NEW" | "EDIT";

export interface CFAnalysisExecution {
  executionId: string;
  counterfactualId: string;
  maxRunningTimeSeconds: number;
}

export interface CFAnalysisResult {
  executionId: string;
  counterfactualId: string;
  type: "counterfactual";
  valid: boolean;
  status: "SUCCEEDED" | "FAILED";
  statusDetails: string;
  solutionId: string;
  isValid: boolean;
  stage: "INTERMEDIATE" | "FINAL";
  inputs: ItemObject[];
  outputs: ItemObject[];
  sequenceId: number;
}

export interface CFAnalysisResultsSets extends CFAnalysisExecution {
  goals: CFGoal[];
  searchDomains: CFSearchInput[];
  solutions: CFAnalysisResult[];
}

export type CFSupportMessage = {
  id: string;
  message: string;
};
