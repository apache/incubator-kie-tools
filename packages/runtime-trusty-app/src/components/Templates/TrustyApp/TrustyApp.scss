.trusty-logo {
  width: 158px;
}

/*
  Fixing weird toolbar extra spacing introduced after updating to PF 2021.08 (see FAI-611)
  The issue is not present in more recent versions of PF.
  To be removed after the next PF update on kogito-apps.
 */
.pf-c-toolbar__content[hidden] {
  display: none;
}
