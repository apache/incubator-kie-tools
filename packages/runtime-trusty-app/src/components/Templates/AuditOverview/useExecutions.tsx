/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import { useCallback, useContext, useEffect, useMemo, useState } from "react";
import { callOnceHandler, EXECUTIONS_PATH } from "../../../utils/api/httpClient";
import { Executions, RemoteData, RemoteDataStatus } from "../../../types";
import axios, { AxiosRequestConfig } from "axios";
import { TrustyContext } from "../TrustyApp/TrustyApp";

type useExecutionsParameters = {
  searchString: string;
  from: string;
  to: string;
  limit: number;
  offset: number;
};

const useExecutions = (parameters: useExecutionsParameters) => {
  const { searchString, from, to, limit, offset } = parameters;
  const [executions, setExecutions] = useState<RemoteData<Error, Executions>>({
    status: RemoteDataStatus.NOT_ASKED,
  });

  const baseUrl = useContext(TrustyContext).config.serverRoot;

  const getExecutions = useMemo(() => callOnceHandler(), []);

  const loadExecutions = useCallback(() => {
    let isMounted = true;
    setExecutions({ status: RemoteDataStatus.LOADING });

    const config: AxiosRequestConfig = {
      baseURL: baseUrl,
      url: EXECUTIONS_PATH,
      method: "get",
      params: { search: searchString, from, to, limit, offset },
    };

    getExecutions(config)
      .then((response) => {
        if (isMounted) {
          setExecutions({
            status: RemoteDataStatus.SUCCESS,
            data: response.data,
          });
        }
      })
      .catch((error) => {
        if (!axios.isCancel(error)) {
          setExecutions({ status: RemoteDataStatus.FAILURE, error });
        }
      });
    return () => {
      isMounted = false;
    };
  }, [searchString, from, to, limit, offset]);

  useEffect(() => {
    loadExecutions();
  }, [searchString, from, to, limit, offset, loadExecutions]);

  return { loadExecutions, executions };
};

export default useExecutions;
