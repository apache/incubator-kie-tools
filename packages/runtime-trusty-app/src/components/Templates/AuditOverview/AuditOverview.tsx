/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React, { useEffect, useState } from "react";
import { PageSection, PageSectionVariants, Text, TextContent, Title } from "@patternfly/react-core/dist/js/components";
import { AuditToolbarTop, AuditToolbarBottom } from "../../Organisms/AuditToolbar/AuditToolbar";
import ExecutionTable from "../../Organisms/ExecutionTable/ExecutionTable";
import useExecutions from "./useExecutions";
import { formatISO, sub } from "date-fns";
import { RemoteDataStatus } from "../../../types";

type AuditOverviewProps = {
  dateRangePreset?: {
    fromDate: string;
    toDate: string;
  };
};

const AuditOverview = (props: AuditOverviewProps) => {
  const { dateRangePreset } = props;
  const toPreset = dateRangePreset ? dateRangePreset.toDate : formatISO(Date.now(), { representation: "date" });
  const fromPreset = dateRangePreset
    ? dateRangePreset.fromDate
    : formatISO(sub(Date.now(), { months: 1 }), { representation: "date" });
  const [searchString, setSearchString] = useState("");
  const [fromDate, setFromDate] = useState(fromPreset);
  const [toDate, setToDate] = useState(toPreset);
  const [page, setPage] = useState(1);
  const [pageSize, setPageSize] = useState(10);
  const [total, setTotal] = useState(0);
  const { loadExecutions, executions } = useExecutions({
    searchString,
    from: fromDate,
    to: toDate,
    limit: pageSize,
    offset: pageSize * (page - 1),
  });

  useEffect(() => {
    if (executions.status === RemoteDataStatus.SUCCESS) {
      setTotal(executions.data.total);
    }
  }, [executions]);

  return (
    <>
      <PageSection variant={PageSectionVariants.light}>
        <TextContent>
          <Title size="3xl" headingLevel="h2">
            Audit investigation
          </Title>
          <Text component="p">Here you can retrieve all the available information about past cases.</Text>
        </TextContent>
      </PageSection>

      <PageSection isFilled={true}>
        <AuditToolbarTop
          setSearchString={setSearchString}
          fromDate={fromDate}
          setFromDate={setFromDate}
          toDate={toDate}
          setToDate={setToDate}
          total={total}
          pageSize={pageSize}
          page={page}
          setPage={setPage}
          setPageSize={setPageSize}
          onRefresh={loadExecutions}
        />

        <ExecutionTable data={executions} />

        <AuditToolbarBottom total={total} pageSize={pageSize} page={page} setPage={setPage} setPageSize={setPageSize} />
      </PageSection>
    </>
  );
};

export default AuditOverview;
