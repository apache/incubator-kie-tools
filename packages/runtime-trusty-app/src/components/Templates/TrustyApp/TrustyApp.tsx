/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React, { useContext, useState } from "react";
import { BrowserRouter, NavLink, Redirect, Route, Switch, useLocation } from "react-router-dom";
import {
  Avatar,
  Brand,
  Nav,
  NavItem,
  NavList,
  Page,
  PageHeader,
  PageHeaderTools,
  PageSidebar,
} from "@patternfly/react-core/dist/js/components";
import AuditOverview from "../AuditOverview/AuditOverview";
import kogitoLogo from "../../../static/images/kogitoLogo.svg";
import AuditDetail from "../AuditDetail/AuditDetail";
import imgAvatar from "../../../static/images/user.svg";
import Breadcrumbs from "../../Organisms/Breadcrumbs/Breadcrumbs";
import NotFound from "../NotFound/NotFound";
import ApplicationError from "../ApplicationError/ApplicationError";
import { TrustyContextValue } from "../../../types";
import { datePickerSetup } from "../../Molecules/DatePicker/DatePicker";
// manually importing Red Hat Mono font because not yet provided by PatternFly
// see: https://github.com/patternfly/patternfly/issues/4021
// when updating PF to a version that will include it, it will be
// important to remove this duplication
import "../../../static/fonts/RedHatMono/RedHatMono.css";
import "./TrustyApp.scss";

datePickerSetup();

type TrustyAppProps = {
  /** Enable counterfactual analysis feature */
  counterfactualEnabled: boolean;
  /** Enable explainability information inside decisions */
  explanationEnabled: boolean;
  /** Configuration options to embed Trusty inside a parent application  */
  containerConfiguration?: ContainerConfigurationProps;
};

type ContainerConfigurationProps = {
  /** Include the page layout wrapper with sidebar navigation and breadcrumbs */
  pageWrapper?: boolean;
  /** Use an optional Server root for all REST operations */
  serverRoot: string;
  /** Use an optional base path for internal routes */
  basePath?: string;
  /**
   * Do not include a rect router (BrowserRouter) within TrustyApp.
   * The host application will have to include one in order to make
   * Trusty internal Routes work. */
  excludeReactRouter?: boolean;
  /**
   * Use traditional anchor links with href attributes. When set to false
   * all links will be handled with a onClick handler pushing url
   * to the browser history */
  useHrefLinks?: boolean;
};

const TrustyApp: React.FC<TrustyAppProps> = (props) => {
  const baseUrl = window.TRUSTY_ENDPOINT || process.env.KOGITO_TRUSTY_API_HTTP_URL!;

  const {
    counterfactualEnabled,
    explanationEnabled,
    containerConfiguration: {
      pageWrapper = true,
      serverRoot = baseUrl,
      basePath = "",
      excludeReactRouter = false,
      useHrefLinks = true,
    } = {},
  } = props;

  return (
    <TrustyContext.Provider
      value={{
        config: {
          counterfactualEnabled,
          explanationEnabled,
          serverRoot: serverRoot,
          basePath: basePath,
          useHrefLinks,
        },
      }}
    >
      <>
        {excludeReactRouter ? (
          <TrustyAppContainer pageWrapper={pageWrapper} />
        ) : (
          <BrowserRouter>
            <TrustyAppContainer pageWrapper={pageWrapper} />
          </BrowserRouter>
        )}
      </>
    </TrustyContext.Provider>
  );
};

export default TrustyApp;

export const TrustyContext = React.createContext<TrustyContextValue>({} as TrustyContextValue);

type TrustyAppContainerProps = {
  pageWrapper: boolean;
};

const TrustyAppContainer = ({ pageWrapper }: TrustyAppContainerProps) => {
  const location = useLocation();
  const [isMobileView, setIsMobileView] = useState(false);
  const [isNavOpenDesktop, setIsNavOpenDesktop] = useState(true);
  const [isNavOpenMobile, setIsNavOpenMobile] = useState(false);
  const { config } = useContext(TrustyContext);

  const onNavToggleDesktop = () => {
    setIsNavOpenDesktop(!isNavOpenDesktop);
  };

  const onNavToggleMobile = () => {
    setIsNavOpenMobile(!isNavOpenMobile);
  };
  const handlePageResize = (props: { windowSize: number; mobileView: boolean }) => {
    // closing sidebar menu when resolution is < 1200
    if (props.windowSize < 1200) {
      if (!isMobileView) setIsMobileView(true);
    } else {
      if (isMobileView) setIsMobileView(false);
    }
  };

  const PageNav = (
    <Nav aria-label="Nav" theme="dark">
      <NavList>
        <NavItem isActive={location.pathname.startsWith("/audit")} ouiaId="audit-item">
          <NavLink to="/audit">Audit investigation</NavLink>
        </NavItem>
      </NavList>
    </Nav>
  );

  const Sidebar = (
    <PageSidebar nav={PageNav} isNavOpen={isMobileView ? isNavOpenMobile : isNavOpenDesktop} theme="dark" />
  );

  const Header = (
    <PageHeader
      logo={<Brand src={kogitoLogo} alt="Kogito TrustyAI" className="trusty-logo" />}
      logoProps={{ href: `${config.basePath}/`, className: "trusty-home-link" }}
      headerTools={
        <PageHeaderTools>
          <Avatar src={imgAvatar} alt="Avatar image" />
        </PageHeaderTools>
      }
      showNavToggle={isMobileView}
      onNavToggle={isMobileView ? onNavToggleMobile : onNavToggleDesktop}
      isNavOpen={isMobileView ? isNavOpenMobile : isNavOpenDesktop}
    />
  );

  const Routes = (
    <Switch>
      <Route exact path={`${config.basePath}/`}>
        <Redirect to={`${config.basePath}/audit`} />
      </Route>
      <Route exact path={`${config.basePath}/audit`}>
        <AuditOverview />
      </Route>
      <Route path={`${config.basePath}/audit/:executionType/:executionId`}>
        <AuditDetail />
      </Route>
      <Route exact path="/error">
        <ApplicationError />
      </Route>
      <Route path="/not-found" component={NotFound} />
      <Redirect to="/not-found" />
    </Switch>
  );

  return (
    <>
      {pageWrapper ? (
        <Page header={Header} sidebar={Sidebar} breadcrumb={<Breadcrumbs />} onPageResize={handlePageResize}>
          {Routes}
        </Page>
      ) : (
        <>{Routes}</>
      )}
    </>
  );
};
