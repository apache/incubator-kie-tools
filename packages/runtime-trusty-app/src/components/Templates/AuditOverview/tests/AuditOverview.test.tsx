/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React from "react";
import { mount, shallow } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import AuditOverview from "../AuditOverview";
import useExecutions from "../useExecutions";
import { RemoteDataStatus } from "../../../../types";
import { TrustyContext } from "../../TrustyApp/TrustyApp";

jest.mock("../useExecutions");

describe("Audit overview", () => {
  test("renders correctly a list of executions", () => {
    const mockLoadExecutions = jest.fn();
    const executions = {
      status: RemoteDataStatus.SUCCESS,
      data: {
        total: 1,
        limit: 10,
        offset: 0,
        headers: [
          {
            executionId: "b2b0ed8d-c1e2-46b5-ad4f-3ac54ff4beae",
            executionDate: "2020-06-01T12:33:57+0000",
            executionSucceeded: true,
            executorName: "testUser",
            executedModelName: "LoanEligibility",
            executionType: "DECISION",
          },
        ],
      },
    };
    (useExecutions as jest.Mock).mockReturnValue({
      mockLoadExecutions,
      executions,
    });
    const wrapper = shallow(<AuditOverview dateRangePreset={{ fromDate: "2020-01-01", toDate: "2020-02-01" }} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("loads a list of executions from the last month by default", () => {
    const mockLoadExecutions = jest.fn();
    const executions = {
      status: RemoteDataStatus.SUCCESS,
      data: {
        total: 1,
        limit: 10,
        offset: 0,
        headers: [
          {
            executionId: "b2b0ed8d-c1e2-46b5-ad4f-3ac54ff4beae",
            executionDate: "2020-06-01T12:33:57+0000",
            executionSucceeded: true,
            executorName: "testUser",
            executedModelName: "LoanEligibility",
            executionType: "DECISION",
          },
        ],
      },
    };
    (useExecutions as jest.Mock).mockReturnValue({
      mockLoadExecutions,
      executions,
    });
    const fixedDate = "2020-05-01T00:00:00.000Z";

    jest.spyOn(global.Date, "now").mockImplementation(() => new Date(fixedDate).getTime());

    const wrapper = mount(
      <TrustyContext.Provider
        value={{
          config: {
            counterfactualEnabled: true,
            explanationEnabled: true,
            basePath: "",
            useHrefLinks: true,
          },
        }}
      >
        <MemoryRouter>
          <AuditOverview />
        </MemoryRouter>
      </TrustyContext.Provider>
    );

    expect(useExecutions).toHaveBeenCalledWith({
      searchString: "",
      from: "2020-04-01",
      to: "2020-05-01",
      limit: 10,
      offset: 0,
    });
    expect(wrapper.find("ExecutionTable").props().data).toStrictEqual(executions);
  });
});
