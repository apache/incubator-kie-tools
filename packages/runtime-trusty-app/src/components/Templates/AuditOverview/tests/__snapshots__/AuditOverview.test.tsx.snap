// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Audit overview renders correctly a list of executions 1`] = `
<Fragment>
  <PageSection
    variant="light"
  >
    <TextContent>
      <Title
        headingLevel="h2"
        size="3xl"
      >
        Audit investigation
      </Title>
      <Text
        component="p"
      >
        Here you can retrieve all the available information about past cases.
      </Text>
    </TextContent>
  </PageSection>
  <PageSection
    isFilled={true}
  >
    <AuditToolbarTop
      fromDate="2020-01-01"
      page={1}
      pageSize={10}
      setFromDate={[Function]}
      setPage={[Function]}
      setPageSize={[Function]}
      setSearchString={[Function]}
      setToDate={[Function]}
      toDate="2020-02-01"
      total={0}
    />
    <ExecutionTable
      data={
        Object {
          "data": Object {
            "headers": Array [
              Object {
                "executedModelName": "LoanEligibility",
                "executionDate": "2020-06-01T12:33:57+0000",
                "executionId": "b2b0ed8d-c1e2-46b5-ad4f-3ac54ff4beae",
                "executionSucceeded": true,
                "executionType": "DECISION",
                "executorName": "testUser",
              },
            ],
            "limit": 10,
            "offset": 0,
            "total": 1,
          },
          "status": 3,
        }
      }
    />
    <AuditToolbarBottom
      page={1}
      pageSize={10}
      setPage={[Function]}
      setPageSize={[Function]}
      total={0}
    />
  </PageSection>
</Fragment>
`;
